File: app/lib/src/domain/models/batch_rabbit.dart
class BatchRabbitItem {
  const BatchRabbitItem({
    required this.id,
    required this.batchId,
    required this.rabbitId,
    required this.currentStatus,
    required this.nextEventType,
    this.maleRabbitId,
    this.latestCycleId,
    this.currentNursingKits = 0,
    this.nursingLitterCount = 0,
    this.joinReason = '',
    this.batchRole = '',
    this.lastEventDate,
    this.nextEventDate,
    this.isActive = true,
    this.joinDate,
    this.exitDate,
    this.rabbitType = '',
    this.rabbitGender = '',
    this.cageId,
  });

  final int id;
  final int batchId;
  final int rabbitId;
  final String currentStatus;
  final String nextEventType;
  final int? maleRabbitId;
  final int? latestCycleId;
  final int currentNursingKits;
  final int nursingLitterCount;
  final String joinReason;
  final String batchRole;
  final DateTime? lastEventDate;
  final DateTime? nextEventDate;
  final bool isActive;
  final DateTime? joinDate;
  final DateTime? exitDate;
  final String rabbitType;
  final String rabbitGender;
  final int? cageId;

  bool get isNursing => currentNursingKits > 0;

  static BatchRabbitItem fromJson(Map<String, dynamic> json) {
    return BatchRabbitItem(
      id: _intValue(json['id']),
      batchId: _intValue(json['batchId']),
      rabbitId: _intValue(json['rabbitId']),
      currentStatus: json['currentStatus'] as String? ?? '',
      nextEventType: json['nextEventType'] as String? ?? '',
      maleRabbitId: _nullableInt(json['maleRabbitId']),
      latestCycleId: _nullableInt(json['latestCycleId']),
      currentNursingKits: _intValue(json['currentNursingKits']),
      nursingLitterCount: _intValue(json['nursingLitterCount']),
      joinReason: json['joinReason'] as String? ?? '',
      batchRole: json['batchRole'] as String? ?? '',
      lastEventDate: _parseDate(json['lastEventDate']),
      nextEventDate: _parseDate(json['nextEventDate']),
      isActive: _boolValue(json['isActive'], fallback: true),
      joinDate: _parseDate(json['joinDate']),
      exitDate: _parseDate(json['exitDate']),
      rabbitType: json['rabbitType'] as String? ?? '',
      rabbitGender: json['rabbitGender'] as String? ?? '',
      cageId: _nullableInt(json['cageId']),
    );
  }

  static int? _nullableInt(Object? value) {
    final parsed = _intValue(value);
    return parsed <= 0 ? null : parsed;
  }

  static DateTime? _parseDate(Object? value) {
    if (value is String && value.isNotEmpty) {
      return DateTime.tryParse(value);
    }
    if (value is num) {
      return DateTime.fromMillisecondsSinceEpoch(value.toInt());
    }
    return null;
  }

  static bool _boolValue(Object? value, {required bool fallback}) {
    if (value is bool) {
      return value;
    }
    if (value is num) {
      return value != 0;
    }
    if (value is String) {
      if (value.toLowerCase() == 'true' || value == '1') {
        return true;
      }
      if (value.toLowerCase() == 'false' || value == '0') {
        return false;
      }
    }
    return fallback;
  }

  static int _intValue(Object? value) {
    if (value is int) {
      return value;
    }
    if (value is num) {
      return value.toInt();
    }
    if (value is String) {
      return int.tryParse(value) ?? 0;
    }
    return 0;
  }
}

File: backend/rabbit-production/src/main/java/com/rabbit/app/modules/batch/entity/BatchRabbit.java
package com.rabbit.app.modules.batch.entity;

import java.util.Date;

public class BatchRabbit {
    private Long id;
    private Long batchId;
    private Long rabbitId;
    private Long maleRabbitId;
    private Long latestCycleId;
    private Integer currentNursingKits;
    private Integer nursingLitterCount;
    private String joinReason;
    private String batchRole;
    private String currentStatus;
    private Date lastEventDate;
    private Date nextEventDate;
    private String nextEventType;
    private Boolean isEventNotified;
    private Date eventNotifyDate;
    private Boolean isActive;
    private Date joinDate;
    private Date exitDate;
    private String remark;
    private String createBy;
    private Date createTime;
    private String updateBy;
    private Date updateTime;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getBatchId() {
        return batchId;
    }

    public void setBatchId(Long batchId) {
        this.batchId = batchId;
    }

    public Long getRabbitId() {
        return rabbitId;
    }

    public void setRabbitId(Long rabbitId) {
        this.rabbitId = rabbitId;
    }

    public Long getMaleRabbitId() {
        return maleRabbitId;
    }

    public void setMaleRabbitId(Long maleRabbitId) {
        this.maleRabbitId = maleRabbitId;
    }

    public Long getLatestCycleId() {
        return latestCycleId;
    }

    public void setLatestCycleId(Long latestCycleId) {
        this.latestCycleId = latestCycleId;
    }

    public Integer getCurrentNursingKits() {
        return currentNursingKits;
    }

    public void setCurrentNursingKits(Integer currentNursingKits) {
        this.currentNursingKits = currentNursingKits;
    }

    public Integer getNursingLitterCount() {
        return nursingLitterCount;
    }

    public void setNursingLitterCount(Integer nursingLitterCount) {
        this.nursingLitterCount = nursingLitterCount;
    }

    public String getJoinReason() {
        return joinReason;
    }

    public void setJoinReason(String joinReason) {
        this.joinReason = joinReason;
    }

    public String getBatchRole() {
        return batchRole;
    }

    public void setBatchRole(String batchRole) {
        this.batchRole = batchRole;
    }

    public String getCurrentStatus() {
        return currentStatus;
    }

    public void setCurrentStatus(String currentStatus) {
        this.currentStatus = currentStatus;
    }

    public Date getLastEventDate() {
        return lastEventDate;
    }

    public void setLastEventDate(Date lastEventDate) {
        this.lastEventDate = lastEventDate;
    }

    public Date getNextEventDate() {
        return nextEventDate;
    }

    public void setNextEventDate(Date nextEventDate) {
        this.nextEventDate = nextEventDate;
    }

    public String getNextEventType() {
        return nextEventType;
    }

    public void setNextEventType(String nextEventType) {
        this.nextEventType = nextEventType;
    }

    public Boolean getIsEventNotified() {
        return isEventNotified;
    }

    public void setIsEventNotified(Boolean eventNotified) {
        isEventNotified = eventNotified;
    }

    public Date getEventNotifyDate() {
        return eventNotifyDate;
    }

    public void setEventNotifyDate(Date eventNotifyDate) {
        this.eventNotifyDate = eventNotifyDate;
    }

    public Boolean getIsActive() {
        return isActive;
    }

    public void setIsActive(Boolean active) {
        isActive = active;
    }

    public Date getJoinDate() {
        return joinDate;
    }

    public void setJoinDate(Date joinDate) {
        this.joinDate = joinDate;
    }

    public Date getExitDate() {
        return exitDate;
    }

    public void setExitDate(Date exitDate) {
        this.exitDate = exitDate;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }

    public String getCreateBy() {
        return createBy;
    }

    public void setCreateBy(String createBy) {
        this.createBy = createBy;
    }

    public Date getCreateTime() {
        return createTime;
    }

    public void setCreateTime(Date createTime) {
        this.createTime = createTime;
    }

    public String getUpdateBy() {
        return updateBy;
    }

    public void setUpdateBy(String updateBy) {
        this.updateBy = updateBy;
    }

    public Date getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(Date updateTime) {
        this.updateTime = updateTime;
    }
}
